openedx/core/djangoapps/catalog/management/commands/tests/test_cache_programs.py

```diff
@@ -269,6 +322,98 @@ def test_pathways_multiple_pages(self):
 
             self.assertEqual(pathway, pathways_dict[key])
 
+    
+    def test_handle_courses(self):
+        """
+        Verify that the command requests and caches course to program uuids
+        """
+
+        UserFactory(username=self.catalog_integration.service_username)
+
+        programs = {
+            PROGRAM_CACHE_KEY_TPL.format(uuid=program['uuid']): program for program in self.programs
+        }
+
+        pathways = {
+            PATHWAY_CACHE_KEY_TPL.format(id=pathway['id']): pathway for pathway in self.pathways
+        }
+
+        courses = {
+            COURSE_PROGRAMS_CACHE_KEY_TPL.format(course_run_id=course['key']):
+            [pu['uuid'] for pu in course['programs']]
+            for course in self.course_runs
+        }
+
+        self.mock_list()
+        self.mock_pathways(self.pathways)
+        self.mock_courses(self.course_runs)
+
+        for uuid in self.uuids:
+            program = programs[PROGRAM_CACHE_KEY_TPL.format(uuid=uuid)]
+            self.mock_detail(uuid, program)
+
+        call_command('cache_programs')
+
+        cached_courses = cache.get_many(list(courses.keys()))
+        self.assertEqual(
+            set(cached_courses),
+            set(courses)
+        )
+
+        # We can't use a set comparison here because these values are dictionaries
+        # and aren't hashable. We've already verified that all pathways came out
+        # of the cache above, so all we need to do here is verify the accuracy of
+        # the data itself.
+        for key, course in cached_courses.items():
+            self.assertEqual(course, courses[key])
+
+    def test_handle_multiple_courses(self):
+        """
+        Verify that the command requests and caches course to program uuids
```

Comment:
clarify difference between ```test_handle_courses``` and ```test_handle_multiple_courses``` (either in the docstring or in the test name)

Reply:
Good catch.